app/integration_action.go

```diff
@@ -69,13 +69,6 @@ func (a *App) DoPostActionWithCookie(postId, actionId, userId, selectedOption st
 		close(pchan)
 	}()
 
-	cchan := make(chan store.StoreResult, 1)
```

Comment:
This code was here to make the db requests asynchronous and improve performance. I would like to keep it here, or is there a reason not to?

Reply:
This requests the channel from the postId but when the post does not exist, it's useless.
It was not a problem before, but as I need the channel to get the teamId, I would have to make another synchronous request when the post does not exist (to get the channel from the cookie's channelId line 85).
With this change I'm doing only one request to get the channel but if you prefer, I can change it and do another request just when using the cookie.
Just let me know.

Reply:
The usual case is the the cookie is not set and we jump into the else case of L94. Hence, It don't see a problem with just making the request even though we might not need it.

Reply:
Ok, I didn't know this was the usual case but with this in mind, I'll rework the PR. Thanks

Reply:
I changed the requests flow in 1b85d42, just let me know what you think.